composer/utils/object_store/s3_object_store.py

```diff
@@ -99,7 +102,7 @@ def __init__(
         self.transfer_config = TransferConfig(**transfer_config)
 
     def get_uri(self, object_name: str) -> str:
-        return f's3://{self.bucket}/{object_name}'
+        return os.path.join('s3://', self.bucket, self.prefix, object_name)
```

Comment:
Since object store paths must use forward slashes, would prefer to use manual string building for the URI rather than `os.path.join`, which could result in a backwards slash (`\`) on windows. If you want to use a helper, take a look at https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunsplit